include/mesh.tcc

```diff
@@ -1883,3 +1883,36 @@ bool mpm::Mesh<Tdim>::assign_nodal_friction_constraints(
   }
   return status;
 }
+
+//! Assign nodal velocity constraints
+template <unsigned Tdim>
+bool mpm::Mesh<Tdim>::initialise_nodal_properties(int set_id) {
+  bool status = true;
+  // Initialise the shared pointer to nodal properties
+  nodal_properties_ = std::make_shared<mpm::NodalProperties>();
+  try {
+    if (set_id == -1 || node_sets_.find(set_id) != node_sets_.end()) {
```

Comment:
If we have a dynamic interface, then the nodal property size are going to change all the time. So, I'd suggest we set the nodal properties to all the nodes initially, we can improve this later. I wouldn't want to pass a set-id because it doesn't get updated at each time step.

Reply:
I agree. I only added the option of choosing a set because I though it could be useful for initial testing. I will just leave the initialise function with all the nodes, though, as you instructed.